step-capstone/src/components/Trip.js

```diff
@@ -0,0 +1,247 @@
+        departureDate: "4:00pm EST",
+        arrivalDate: "7:00pm PST",
+        description: "Additional notes"
+    },
+    {
+        id: 100,
+        finalized: true,
+        type: "hotel",
+        name: "Hotel ZED",
+        location: "London",
+        checkIn: "December 17, 2020 15:30:00",
+        checkOut: "December 20, 2020 11:00:00",
+        description: 'description',
+    }
+]
+
+export default class Trip extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            items: []
+        }
+
+        this.handleRemoveItem = this.handleRemoveItem.bind(this);
+        this.handleEditItem = this.handleEditItem.bind(this);
+        this.handleAddItem = this.handleAddItem.bind(this);
+    }
+
+    fetchData() {
+        return testData;
+    }
+
+    componentDidMount() {
+        let data = this.fetchData();
+        this.setState({
+            items: data
+        })
+    }
+
+    handleRemoveItem(id) {
+        console.log("removing" + id)
+        this.setState({
+            items: this.state.items.filter((item) => item.id !== id)
+        });
+    }
+
+    handleEditItem(id) {
+        console.log("editing " + id)
+    }
+
+    handleAddItem(type, data) {
+        if (data === undefined) {
+            console.log("please enter information")
+        } else {
+            // Add to database here
+            // current code for testing data format etc.
+            switch (type) {
+                case "event":
+                    console.log("adding event");
+                    break;
+                case "hotel":
+                    this.setState({
+                        items: this.state.items.concat([{
+                            id: this.state.items.length,
+                            finalized: data.finalized,
+                            type: "hotel",
+                            name: data.title,
+                            location: data.location,
+                            checkIn: data.startDate,
+                            checkOut: data.endDate,
+                            description: data.description,
+
+                        }])
+                    })
+                    break;
+                case "flight":
+                    this.setState({
+                        items: this.state.items.concat([{
+                            id: this.state.items.length,
+                            finalized: data.finalized,
+                            type: type,
+                            departureAirport: data.departureAirport,
+                            arrivalAirport: data.arrivalAirport,
+                            departureDate: data.departureDate.toString(),
+                            arrivalDate: data.arrivalDate.toString(),
+                            description: data.description
+                        }])
+                    })
+                    break;
+                default: console.log("invalid type"); break;
```

Comment:
If this happens, then your code has done something very unexpected. you should probably do something more severe than logging, at least a warning and possibly an error.